trajopt_optimizers/trajopt_sqp/test/joint_position_optimization_unit.cpp

```diff
@@ -0,0 +1,133 @@
+﻿#include <trajopt_utils/macros.h>
+TRAJOPT_IGNORE_WARNINGS_PUSH
+#include <gtest/gtest.h>
+#include <iostream>
+
+#include <ifopt/problem.h>
+#include <ifopt/ipopt_solver.h>
+#include <console_bridge/console.h>
+TRAJOPT_IGNORE_WARNINGS_POP
+
+#include <trajopt_sqp/trust_region_sqp_solver.h>
+#include <trajopt_sqp/osqp_eigen_solver.h>
+#include <trajopt_ifopt/constraints/joint_position_constraint.h>
+#include <trajopt_ifopt/variable_sets/joint_position_variable.h>
+
+const bool DEBUG = false;
+
+class JointPositionOptimization : public testing::TestWithParam<const char*>
+{
+public:
+  // 1) Create the problem
+  ifopt::Problem nlp_;
+
+  void SetUp() override
+  {
+    if (DEBUG)
+      console_bridge::setLogLevel(console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_DEBUG);
+    else
+      console_bridge::setLogLevel(console_bridge::LogLevel::CONSOLE_BRIDGE_LOG_NONE);
+
+    // 1) Create the problem
+    ifopt::Problem nlp;
+
+    // 2) Add Variables
+    std::vector<trajopt::JointPosition::Ptr> vars;
+    for (int ind = 0; ind < 2; ind++)
+    {
+      auto pos = Eigen::VectorXd::Zero(7);
+      std::vector<std::string> joint_names(7, "name");
+      auto var = std::make_shared<trajopt::JointPosition>(pos, joint_names, "Joint_Position_" + std::to_string(ind));
+      vars.push_back(var);
+      nlp_.AddVariableSet(var);
+    }
+
+    // 3) Add constraints
+    Eigen::VectorXd start_pos = Eigen::VectorXd::Zero(7);
+    std::vector<trajopt::JointPosition::Ptr> start;
+    start.push_back(vars.front());
+
+    auto start_constraint = std::make_shared<trajopt::JointPosConstraint>(start_pos, start, "StartPosition");
+    nlp_.AddConstraintSet(start_constraint);
+
+    Eigen::VectorXd end_pos = Eigen::VectorXd::Ones(7);
+    std::vector<trajopt::JointPosition::Ptr> end;
+    end.push_back(vars.back());
+    auto end_constraint = std::make_shared<trajopt::JointPosConstraint>(end_pos, end, "EndPosition");
+    nlp_.AddConstraintSet(end_constraint);
+
+    if (DEBUG)
+    {
+      nlp_.PrintCurrent();
+      std::cout << "Jacobian: \n" << nlp_.GetJacobianOfConstraints() << std::endl;
+    }
+  }
+};
+
+/**
+ * @brief Applies a joint position constraint and solves the problem with IPOPT
+ */
+TEST_F(JointPositionOptimization, joint_position_optimization_ipopt)
+{
+  ifopt::Problem nlp_ipopt(nlp_);
+  ifopt::IpoptSolver solver;
+  solver.SetOption("derivative_test", "first-order");
+  solver.SetOption("linear_solver", "mumps");
+  //  ipopt.SetOption("jacobian_approximation", "finite-difference-values");
+  solver.SetOption("jacobian_approximation", "exact");
```

Comment:
Should the commented code be removed or have a TODO?

Reply:
I left it in there to show how to turn numeric differentiation on since these are the only examples we have.

Reply:
sounds good